backend/src/main/java/com/bakdata/conquery/models/events/CBlock.java

```diff
@@ -101,65 +105,78 @@ public CDateRange getEntityDateRange(int entity) {
 	}
 
 	public int getEntityMinDate(int entity) {
-		return minDate.getOrDefault(entity, Integer.MIN_VALUE);
+		return minDate[getEntityIndex(entity)];
 	}
 
 	public int getEntityMaxDate(int entity) {
-		return maxDate.getOrDefault(entity, Integer.MAX_VALUE);
+		return maxDate[getEntityIndex(entity)];
 	}
 
 	@Override
 	@JsonIgnore
 	public CBlockId createId() {
-		return new CBlockId(bucket, connector);
+		return new CBlockId(bucket.getId(), connector);
 	}
 
 	public void initIndizes(int bucketSize) {
-		includedConcepts = new Int2LongArrayMap(bucketSize);
-		includedConcepts.defaultReturnValue(0);
+		root = bucket.getBucket() * bucketSize;
+
+		includedConcepts = new long[bucketSize];
+
+		minDate = new int[bucketSize];
+		maxDate = new int[bucketSize];
 
-		minDate = new Int2IntArrayMap(bucketSize);
-		maxDate = new Int2IntArrayMap(bucketSize);
+		for (int index = 0; index < bucketSize; index++) {
+			includedConcepts[index] = 0;
+
+			minDate[index] = Integer.MIN_VALUE;
+			maxDate[index] = Integer.MAX_VALUE;
+		}
```

Comment:
```suggestion
		Arrays.fill(includedConcepts, 0);
		Arrays.fill(minDate, Integer.MIN_VALUE);
		Arrays.fill(maxDate, Integer.MAX_VALUE);
```
ändert denke ich nichts an der Performance

Reply:
Hattest du nicht sonst mit den gegenüberliegenden extrema gearbeitet, also bei min MAX_VALUE gesetzt um einen Special-Case zu symbolisieren und andersherum

Reply:
Ja aber das hat daten ohne validity date verhindert bzw einen special case draus gemacht.